### ISSUE TITLE
Fullscreen touch conflicts with gestures

### ISSUE THREAD

**Issue by Namek** [TouchScript/TouchScript#90]
I have some objects in my scene which I want to listen on TapGesture or PanGesture. But there's also a situation where I'd like to listen for taps whenever any object having TapGesture component is not tapped. To describe differently - I want to have fullscreen tap event when any other tap was not recognized.

I tried to use Fullscreen Target and make it friends but as I read in topic #56 it couldn't be done without hierarchy where Fullscreen Target is on top of object hierarchy. Since I use UnitySteer I cannot make such hierarchy. Last post in adnoted discussion was written 4 months ago, before TouchScript 5.0.

So is it possible in TouchScript 5.0? I thought that modifying CameraLayer to change sort but... #89

**Comment by mfaizanattique** [comment]
I am facing the same problem. I have couple of object in my 3d scene which has tap events. I also have a Main Camera which has the Fullscreen layer that works fine with panning, rotating and scaling but my object in the scene's tap events are not firing after i add fullscreen layer. I tried the layer solution mentioned above, but in that way the objects events get fired and the camera panning stops! need help.

**Comment by josimard** [comment]
Facing the same problem, I understood that any collider of a CameraLayer will stop the events from getting to the Fullscreen Layer, unless an "Untouchable" behavior is added to that game object.

This makes it very hard to have gestures globally control the camera AND interact with in-game elements, which is the basic needs of any top-down game.

It would be nice to allow events to pass trough all colliders unless a gesture is specified on the object. Maybe a CameraLayer option? I also tought that a Fullscreen Layer should be able to capture all gestures at all time, but I cannot find how...

**Comment by valyard** [comment]
You could set Fullscreen Layer as the top root object of the whole your hierarchy. This way it will capture all touch events.

I don't see how the system would determine that a collider is actually used or should be ignored without a hint from developer.

**Comment by mfaizanattique** [comment]
OK, i will try this technique and let you know. Well it would have been nice if we could pan, rotate,scale the camera and as well as tap on objects along with it.Thanks

**Comment by leoniddrag** [comment]
Did someone solve this problem? I can't believe that it is impossible to implement basic functional for top-down game (move camera and touch objects on map) with this powerful library.

**Comment by bryab** [comment]
I'm also confused here, and cannot find a solution.  Like the last commenter said, the problem is combining full-screen navigation gestures with tapping on 3d object colliders.  No matter what I do, one seems to overtake the other, and they don't work together.

**Comment by valyard** [comment]
@bryab can you describe what objects hierarchy you have and what behavior you want?

**Comment by bryab** [comment]
First, I started with the Touchscript "Camera" example which implements pan, pinch zoom, and rotation for the camera itself, using a "Pivot" for panning and rotating, and the Camera itself for zooming.  it is setup like this:

- CameraPivot [ScreenTransform Gestures] [Fullscreen Layer]
  - Camera [Camera Layer]

Then, I have 3d objects in the scene with colliders and TapGesture implementations.

I want the CameraLayer to "try" 3D colliders first, and when that fails, defer to the FullscreenLayer nav controls.

In other words, when a touch happens, and there is a 3d collider under the touch, and the GameObject of that collider implements a Touchscript gesture, and that gesture "succeeds", then the ray stops there and FullscreenLayer is not used.  However, if there is a 3d collider under the touch but either no Touchscript implementation, or the Touchscript gesture "fails", I want the touch to then continue on to the FullscreenLayer.

**Comment by Matheus-de-Souza** [comment]
Man, I did it!

I made a copy of the script `StandardLayer` as is mentioned [here](https://github.com/TouchScript/TouchScript/wiki/Layers#implementing-your-own-layer) and I renamed the new script as TapObjectsLayer. Them I removed all inputModule mentions in the new script and rewrote the first lines of the method `Hit`

        public override HitResult Hit(IPointer pointer, out HitData hit)
        {
            //TouchScript.Pointers.TouchPointer.get
            if (base.Hit(pointer, out hit) != HitResult.Hit) return HitResult.Miss;

            if (TouchManager.Instance.Pointers.Count > 1)
            {
                return HitResult.Miss;
            }

And it worked nicely! :D 

PS.: I know it's a monkey patch, but it worked for me hehe
